ee/modules/030-cloud-provider-openstack/images/cloud-data-discoverer/discoverer.go

```diff
@@ -84,3 +98,252 @@ func (d *Discoverer) InstanceTypes(_ context.Context) ([]v1alpha1.InstanceType,
 
 	return res, nil
 }
+
+func (d *Discoverer) DiscoveryData(ctx context.Context) ([]byte, error) {
```

Comment:
Consider implementing backoff logic.

https://pkg.go.dev/github.com/gophercloud/gophercloud

@name212

Reply:
@artbin Let implement backoff logic for openstack request